lib/manageiq/active_record_connector.rb

```diff
@@ -0,0 +1,58 @@
+require "active_record"
+unless defined? DatabaseConfigurationPatch
+  require_relative "../patches/database_configuration_patch.rb"
+end
+
+module ManageIQ
+  class ActiveRecordConnector
+    # Helper class for getting the database config without needing Rails
+    #
+    # Makes use of the database_configuration_patch for the most part for
+    # grabbing this from the yaml file in config/database.yml
+    class ConnectionConfig
+      # Set a default method for handling super call in
+      # lib/patches/database_configuration_patch.rb
+      def self.database_configuration
+        if ENV["DATABASE_URL"]
+          {}
+        end
+      end
+
+      class << self
+        prepend DatabaseConfigurationPatch
+      end
+
+      def self.[](env)
+        database_configuration[env]
+      end
+    end
+
+    # Returns true if an ActiveRecord::Base.connection_config is present,
+    # otherwise returns false
+    def self.connection_exists?
+      true if ActiveRecord::Base.connection_config
```

Comment:
This is in response to @jrafanie's [suggestion earlier](https://github.com/ManageIQ/manageiq/pull/15336#discussion_r121200394):

So, this one is a pain, but to put it simply, basically the `.connected?` method eventually calls this:

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb#L414-L416

Which the `@connections` variable is empty until you call `ActiveRecord::Base.connection` to retrieve a connection, and it will create a new one for you.  So basically, this doesn't return anything meaningful until you attempt to make a query against your database connection.

To add to that, `ActiveRecord::Base.connection` will basically calls `.retrieve_connection`:

https://github.com/rails/rails/blob/master/activerecord/lib/active_record/connection_adapters/abstract/connection_pool.rb#L928-L934

Which will raise the same error that we are already rescuing.  So even if we were to do something like this to make use of `ActiveRecord::Base.connected?`:

```ruby
def self.connection_exists?
  ActiveRecord::Base.connection  # get a connection in case one isn't in the connection pool already
  ActiveRecord::Base.connected? || false
end
```

We would still need to rescue from `ActiveRecord::ConnectionNotEstablished` because `ActiveRecord::Base.connection` will throw that error if the pool doesn't exist (haven't established a connection yet), or a connection can't be checked out (probably the same problem).

So basically, what we have here is probably the most terse code possible, even though it seems like the other interface should be preferred... _sigh_...

@jrafanie Let me know if you still want me to use the `.connected?` form, just seems like it is unnecessarily verbose if we do.